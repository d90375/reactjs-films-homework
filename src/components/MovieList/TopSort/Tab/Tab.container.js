import React from "react";
import { useDispatch } from "react-redux";
import { useHistory } from "react-router-dom";

import Tab from "./Tab";
import { fetchPopularData, fetchTopRatedData, fetchUpcomingData } from "../../../../modules/movieListData";

const TABS_INFO = {
  Trending: fetchPopularData(),
  "Top Rated": fetchTopRatedData(),
  "Coming Soon": fetchUpcomingData()
};

const TabContainer = () => {
  const dispatch = useDispatch();
  const history = useHistory();
  const handleSelectTab = (tabName, fetchFunc) => {
    return () => {
      history.push(`/?filter=${tabName.replace(/\s/g, "").toLowerCase()}`);
      dispatch(fetchFunc);
    };
  };

  return (
    <>
      {Object.entries(TABS_INFO).map((currTabObj) => {
        return (
          <Tab key={currTabObj} onSelectTab={handleSelectTab(currTabObj[0], currTabObj[1])}>
            {currTabObj[0]}
          </Tab>
        );
      })}
    </>
  );
};

export default TabContainer;
