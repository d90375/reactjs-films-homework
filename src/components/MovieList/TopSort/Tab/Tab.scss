.titleText {
  background-color: transparent;
  border: none;
  color: #0aaee4;
  cursor: pointer;
  font-family: "HelveticaNeue", serif;
  font-size: 1rem;
  font-weight: 600;
  letter-spacing: 1.12px;
  line-height: 44px;
  outline: none;
  padding: 0 10px;
  text-decoration: none;
  text-transform: capitalize;

  &:focus,
  &:active {
    outline: none;
  }

  &:hover {
    text-shadow: 0 0 10px rgba(10, 174, 228, 1);
  }

  &:nth-child(4) {
    margin-right: 12px;
  }

  &.active {
    border-bottom: 2px solid #0aaee4;
  }

  @media (max-width: 425px) {
    padding: 5px 5px 0 5px;
    font-size: 0.8rem;
    line-height: 20px;
    width: 100%;
  }
}
