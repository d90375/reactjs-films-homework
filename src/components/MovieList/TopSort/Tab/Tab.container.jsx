import React, { useCallback } from "react";
import { useHistory } from "react-router-dom";
import Tab from "./Tab";
import { TABS_INFO } from "../../../../constants";

const TabContainer = () => {
  const history = useHistory();
  const handleSelectTab = useCallback(
    (tabName) => {
      return () => {
        history.push(`?filter=${tabName.replace(/\s/g, "").toLowerCase()}`);
      };
    },
    [history]
  );

  return (
    <>
      {TABS_INFO.map((currTabObj) => {
        return (
          <Tab key={currTabObj} onSelectTab={handleSelectTab(currTabObj)}>
            {currTabObj}
          </Tab>
        );
      })}
    </>
  );
};

export default TabContainer;
