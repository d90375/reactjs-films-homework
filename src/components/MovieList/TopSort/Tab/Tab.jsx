import PropTypes from "prop-types";
import React from "react";
import "./Tab.scss";

const Tab = ({ children, onSelectTab }) => {
  return (
    <>
      <button type="button" onClick={onSelectTab} className="nav__title__text">
        {children}
      </button>
    </>
  );
};

export default Tab;

Tab.propTypes = {
  onSelectTab: PropTypes.func
};

Tab.defaultProps = {
  onSelectTab: () => {}
};
