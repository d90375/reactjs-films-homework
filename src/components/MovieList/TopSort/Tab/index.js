import Tab from "./Tab.container";

export default Tab;
