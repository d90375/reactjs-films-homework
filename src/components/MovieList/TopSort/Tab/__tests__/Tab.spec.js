import React from "react";
import { shallow } from "enzyme";
import Tab from "../Tab";

describe("Tab", () => {
  let component;

  const mockCallBack = jest.fn();

  it("should render TopSort component without props", () => {
    component = shallow(<Tab onSelectTab={mockCallBack} />);
    expect(component).toMatchSnapshot();
  });

  it("should render TopSort component with props, should changed className", () => {
    component = shallow(
      <Tab selectedTab="child" onSelectTab={mockCallBack}>
        child
      </Tab>
    );
    expect(component).toMatchSnapshot();
  });

  describe("handlers", () => {
    it("should call onSelectTab method", () => {
      component = shallow(<Tab onSelectTab={mockCallBack} />);
      expect(mockCallBack.mock.calls.length).toBe(0);
      component.find("button").simulate("click");
      expect(mockCallBack.mock.calls.length).toBe(1);
    });

    it("should use default onSelectTab", () => {
      const result = Tab.defaultProps.onSelectTab();
      expect(result).toBe(undefined);
    });
  });
});
