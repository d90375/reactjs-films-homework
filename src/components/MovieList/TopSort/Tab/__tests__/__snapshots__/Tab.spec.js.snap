// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should render TopSort component with route, should changed className 1`] = `
<Fragment>
  <button
    className="nav__title__text nav__active"
    onClick={[MockFunction]}
    type="button"
  >
    mockChild
  </button>
</Fragment>
`;

exports[`Tab should render TopSort component without props, with fakeChild 1`] = `
<Fragment>
  <button
    className="nav__title__text"
    onClick={[MockFunction]}
    type="button"
  >
    fakeMockChild
  </button>
</Fragment>
`;
